### PR TITLE
Fix test_example_dags

### PR DESCRIPTION
**Pull request by ahidalgob**
This PR solves a couple of bugs in `test_example_dags.py`.

By going up to `parents[3]` we were going outside the repository root, luckily (or unluckily) the repo folder is also named `airflow` so the pattern `airflow/**/example_dags/example_*.py` still worked, but `tests/system/providers/**/example_*.py` wasn't used.

before: collected 105 items
after: collected 695 items 

This discovered 2 new errors:

- `example_local_to_wasb.py` was trivial to fix

- `example_redis_publish.py`: this one fails because `RedisPubSubSensor` constructor calls `Redis.pubsub().subscribe()`, which just hangs and DagBag fails with timeout. For now I'm just deleting this operator from the example.

### CODE REVIEW

**Comment by ahidalgob** [comment]
I think we should move the `subscribe()` call from `RedisPubSubSensor` constructor to the `poke` method. I'm happy to create another PR with this change.

File: tests/always/test_example_dags.py [hunk 1]
```diff
@@ -54,7 +54,9 @@ def example_not_suspended_dags():
     suspended_providers_folders = get_suspended_providers_folders()
     possible_prefixes = ["airflow/providers/", "tests/system/providers/"]
     suspended_providers_folders = [
-        f"{prefix}{provider}" for prefix in possible_prefixes for provider in suspended_providers_folders
+        f"{AIRFLOW_SOURCES_ROOT.as_posix()}/{prefix}{provider}"
```
Comment (uranusjr) [inline tests/always/test_example_dags.py]:
```suggestion
        AIRFLOW_SOURCES_ROOT.joinpath(prefix, provider).as_posix()
```

File: tests/always/test_example_dags.py [hunk 2]
```diff
@@ -68,7 +70,7 @@ def example_dags_except_db_exception():
     return [
         dag_file
         for dag_file in example_not_suspended_dags()
-        if any(not dag_file.endswith(e) for e in NO_DB_QUERY_EXCEPTION)
+        if not any(dag_file.endswith(e) for e in NO_DB_QUERY_EXCEPTION)
```
Comment (uranusjr) [inline tests/always/test_example_dags.py:73]: Fortunately `NO_DB_QUERY_EXCEPTION` has only one entry now so there’s no functional issues…

Comment (ahidalgob) [reply tests/always/test_example_dags.py:73]: Correct, while understanding these failing tests, I added `example_redis_publish.py` for a moment to `NO_DB_QUERY_EXCEPTION` and spotted this bug, but for one element it (luckily(?)) works the same.

**Review by uranusjr — APPROVED** [review]

**Comment by josh-fell** [comment]
@ahidalgob Looks like there is a conflict to resolve. Can you take a look when you have a chance please?

**Review by josh-fell — APPROVED** [review]

**Pull request merged**